lstchain/io/dataselection.py

```diff
@@ -0,0 +1,261 @@
+from ctapipe.core import Component
+from ctapipe.core.traits import Int, Float, List, Dict
+from lstchain.reco.utils import filter_events
+
+import numpy as np
+import astropy.units as u
+from pyirf.binning import create_bins_per_decade  # , add_overflow_bins
+
+__all__ = ["DataSelection", "DataBinning"]
+
+
+class DataSelection(Component):
+    """
+    Collect various selection cuts to be applied for IRF production and
+    DL3 data reduction
+
+    Parameters for event filters will be combined in a dict so that the
+    filter_events() can be used.
+    """
+
+    event_filters = Dict(
+        help="Dict of event filter parameters",
+        default_value={
+            "intensity": [0, np.inf],
+            "length": [0, np.inf],
+            "width": [0, np.inf],
+            "r": [0, 1],
+            "wl": [0.01, 1],
+            "leakage_intensity_width_2": [0, 1],
+        },
+    ).tag(config=True)
+
+    fixed_gh_cut = Float(
+        help="Fixed selection cut for gh_score (gammaness)",
+        default_value=0.6,
+    ).tag(config=True)
+
+    fixed_theta_cut = Float(
+        help="Fixed selection cut for theta",
+        default_value=0.2,
+    ).tag(config=True)
+
+    fixed_source_fov_offset_cut = Float(
+        help="Fixed selection cut for source FoV offset",
+        default_value=2.83,
+    ).tag(config=True)
+
+    lst_tel_ids = List(
+        help="List of selected LST telescope ids",
+        trait=Int(),
+        default_value=[1],
+    ).tag(config=True)
+
+    def filter_cut(self, data):
+        return filter_events(data, self.event_filters)
+
+    def gh_cut(self, data):
+        return data[data["gh_score"] > self.fixed_gh_cut]
+
+    def theta_cut(self, data):
+        return data[data["theta"] < u.Quantity(
+            self.fixed_theta_cut
+            ) * u.deg
+        ]
+
+    def true_src_fov_offset_cut(self, data):
+        return data[
+                data["true_source_fov_offset"] < u.Quantity(
+                    self.fixed_source_fov_offset_cut
+                    ) * u.deg
+            ]
+
+    def reco_src_fov_offset_cut(self, data):
+        return data[
+                data["reco_source_fov_offset"] < u.Quantity(
+                    self.fixed_source_fov_offset_cut
+                    ) * u.deg
+            ]
+
+    def tel_ids_filter(self, data):
+        for i in self.lst_tel_ids:
```

Comment:
Two things here: I don't think this is what you intend, since you don use ` |= ` but `=`, so only the last tel_id in the `tel_ids` is taken.

Then I don't see why this should be called `lst_tel_ids` and not just `allowed_tels` like the traitlet for the `EventSource`.

Reply:
Also, this is the only filter, where you store it in the table.

Reply:
Which is not helpful, since only the entries where it is true are returned.

So I would propose to change `lst_tel_ids` → `allowed_tels`, fix the issue to only use the last telescope by doing:

```
mask = np.ones(len(data), dtype=bool)
for tel_id in self.allowed_tels:
    mask |= data['tel_id'] == tel_id
return data[mask]
```
```

Reply:
Yes, I understand the mistake here. Thank you for the suggestion, I will change it so.